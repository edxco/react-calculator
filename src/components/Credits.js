import React from 'react';

function Credits() {
  return (
    <div className="Credit">
      <figure>
        <blockquote>Done with React and Redux</blockquote>
        <figcaption>By eduardo Baeza</figcaption>
      </figure>
    </div>
  );
}

export default Credits;
