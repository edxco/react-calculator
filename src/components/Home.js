import React from 'react';

function Home() {
  return (
    <div className="Home px-5">
      <h1>Changelog</h1>
      <p className="text-regular changelog-h1">
        All notable changes to this project will be documented in this file.
        <br />
        [1.0.0] - 2021-07-02
        <br />
        <br />
        <strong>Added</strong>
        <br />
        - New visual identity
        <br />
        - Version navigation
        <br />
        - Link to changelog add
        <br />
      </p>
    </div>
  );
}

export default Home;
