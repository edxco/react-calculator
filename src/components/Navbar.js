import React from 'react';
import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <div className="topnav">
      <Link to="/" className="active"><strong>React Calculator</strong></Link>
      <Link to="/" className="">Changelog</Link>
      <Link to="/calculator" className="">Calculator</Link>
      <Link to="/credits" className="">Credits</Link>
    </div>
  );
}

export default Navbar;
