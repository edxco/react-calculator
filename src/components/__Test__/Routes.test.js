import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from '../Navbar';
import Home from '../Home';
import App from '../App';
import Credits from '../Credits';

test('return content from Routes matches snapshot', () => {
  const component = render(
    <BrowserRouter>
      <>
        <Navbar />
        <Switch>
          <Route path="/" component={Home} exact />
          <Route path="/calculator" component={App} />
          <Route path="/credits" component={Credits} />
        </Switch>
      </>
    </BrowserRouter>,
  );
  expect(component.container).toMatchSnapshot();
});

test('content will render without crash', () => {
  const div = document.createElement('div');
  render(
    <BrowserRouter>
      <>
        <Navbar />
        <Switch>
          <Route path="/" component={Home} exact />
          <Route path="/calculator" component={App} />
          <Route path="/credits" component={Credits} />
        </Switch>
      </>
    </BrowserRouter>, div,
  );
});
