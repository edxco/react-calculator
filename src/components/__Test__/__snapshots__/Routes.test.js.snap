// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return content from Routes matches snapshot 1`] = `
<div>
  <div
    class="topnav"
  >
    <a
      class="active"
      href="/"
    >
      <strong>
        React Calculator
      </strong>
    </a>
    <a
      class=""
      href="/"
    >
      Changelog
    </a>
    <a
      class=""
      href="/calculator"
    >
      Calculator
    </a>
    <a
      class=""
      href="/credits"
    >
      Credits
    </a>
  </div>
  <div
    class="Home px-5"
  >
    <h1>
      Changelog
    </h1>
    <p
      class="text-regular changelog-h1"
    >
      All notable changes to this project will be documented in this file.
      <br />
      [1.0.0] - 2021-07-02
      <br />
      <br />
      <strong>
        Added
      </strong>
      <br />
      - New visual identity
      <br />
      - Version navigation
      <br />
      - Link to changelog add
      <br />
    </p>
  </div>
</div>
`;
